### PR TITLE
hostinfo: set Hostinfo.PackageType for mkctr container builds

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes tailscale/corp#21448

### CODE REVIEW

File: hostinfo/packagetype_container.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//gobuild linux && ts_package_container
+
+package hostinfo
+
+func init() {
+	packageType = packageTypeContainer
+}
+
+func packageTypeContainer() string {
+	return "container"
```
Comment (awly) [inline hostinfo/packagetype_container.go:13]: nit: maybe `ts_container` or `official_container`?

**Review by awly — APPROVED** [review]

File: hostinfo/packagetype_container.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//gobuild linux && ts_package_container
+
+package hostinfo
+
+func init() {
+	packageType = packageTypeContainer
```
Comment (awly) [inline hostinfo/packagetype_container.go:9]: oh, actually we probably need to split this up from https://github.com/tailscale/tailscale/blob/60ccd190efc2cc010ac3ad2ce81e0518c7f9b440/hostinfo/hostinfo_linux.go#L21, so `packagetype_linux_nocontainer.go` and `packagetype_linux_container.go`

File: hostinfo/packagetype_container.go (hunk 1 above)
Comment (bradfitz) [reply hostinfo/packagetype_container.go:13]: None of the other values contain "ts_" or "official", though....

Screenshot 2024-07-17 at 10 54 33 AM

File: hostinfo/packagetype_container.go (hunk 2 above)
Comment (bradfitz) [reply hostinfo/packagetype_container.go:9]: whoops

**Review by awly — APPROVED** [review]

**Pull request merged**